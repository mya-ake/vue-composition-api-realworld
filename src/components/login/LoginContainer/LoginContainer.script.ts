import { createComponent, reactive } from '@vue/composition-api';
import { LoginForm, LoginFormType } from './../LoginForm';
import { BaseErrorMessageList } from '@/components/base';
import { useSubmit } from '@/compositions/form';

export default createComponent({
  components: {
    LoginForm,
    BaseErrorMessageList,
  },
  setup() {
    const state = reactive<{
      messages: string[];
    }>({ messages: [] });
    const submitListener = (values: LoginFormType.Values) => {
      // TODO login process
      console.log(values);
    };
    const { disabled, onSubmit } = useSubmit(submitListener);
    return { state, disabled, onSubmit };
  },
});
