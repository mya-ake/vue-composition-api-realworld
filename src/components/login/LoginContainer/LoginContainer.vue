<template>
  <div>
    <BaseErrorMessageList :messages="state.messages" />
    <LoginForm @submit="onSubmit" />
  </div>
</template>

<script lang="ts" src="./LoginContainer.script.ts"></script>
