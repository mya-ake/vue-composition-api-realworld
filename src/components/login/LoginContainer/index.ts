export { default as LoginContainer } from './LoginContainer.vue';
export * from './LoginContainer.type';
