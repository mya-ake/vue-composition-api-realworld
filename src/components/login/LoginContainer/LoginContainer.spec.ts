import { shallowMount, Wrapper } from '@vue/test-utils';
import LoginContainer from './LoginContainer.vue';

describe('components/login/LoginContainer', () => {
  describe('mountable', () => {
    let wrapper: Wrapper<Vue>;

    beforeEach(() => {
      wrapper = shallowMount(LoginContainer, {
        stubs: {
          BaseErrorMessageList: true,
        },
      });
    });

    it('is vue instance', () => {
      expect(wrapper.isVueInstance()).toBe(true);
    });

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
