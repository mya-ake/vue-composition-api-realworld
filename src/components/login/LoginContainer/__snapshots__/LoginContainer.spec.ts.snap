// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/login/LoginContainer mountable snapshot 1`] = `
<div>
  <baseerrormessagelist-stub
    messages=""
  />
   
  <loginform-stub />
</div>
`;
